src/server/game/VoiceChat/VoiceChatChannel.cpp

```diff
@@ -0,0 +1,568 @@
+/*
+ * This file is part of the AzerothCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation; either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "VoiceChatChannel.h"
+#include "BattlegroundMgr.h"
+#include "ChannelMgr.h"
+#include "GroupMgr.h"
+#include "Player.h"
+#include "VoiceChatMgr.h"
+
+VoiceChatChannel::VoiceChatChannel(VoiceChatChannelTypes type, uint32 id, uint32 groupId, std::string name, TeamId team)
+{
+    _channelId = id;
+    _type = type;
+    _channelName = name;
+    _groupId = groupId;
+    _team = team;
+    _isDeleting = false;
+    _isMassAdding = false;
+
+    for (uint8 i = 1; i < MAX_VOICECHAT_CHANNEL_MEMBERS; ++i)
+        _membersGuids[i] = ObjectGuid();
+
+    GenerateSessionId();
+    GenerateEncryptionKey();
+}
+
+VoiceChatChannel::~VoiceChatChannel()
+{
+    _isDeleting = true;
+    for (auto const& m_member : _members)
+        RemoveVoiceChatMember(m_member.second.Guid);
+}
+
+void VoiceChatChannel::GenerateSessionId()
+{
+    // don't know what it should be so use time(nullptr) and increment to make it always different
+    _sessionId = sVoiceChatMgr.GetNewSessionId();
+}
+
+void VoiceChatChannel::GenerateEncryptionKey()
+{
+    // todo figure out encryption key
+    for (auto i = 0; i < 16; i++)
+        _encryptionKey[i] = 0x00;
```

Comment:
Do we understand all the risks involved in releasing it without proper encryption?